enhancements/microshift/generic-device-plugin-for-microshift.md

```diff
@@ -0,0 +1,461 @@
+---
+title: generic-device-plugin-for-microshift
+authors:
+  - pmtk
+reviewers:
+  - "@DanielFroehlich, MicroShift PM"
+  - "@ggiguash, MicroShift contributor"
+  - "@pacevedom, MicroShift contributor"
+  - "@eslutsky, MicroShift contributor"
+  - "@copejon, MicroShift contributor"
+approvers:
+  - "@jerpeter1, MicroShift Staff Engineer"
+api-approvers:
+  - none
+creation-date: 2025-05-13
+last-updated: 2025-05-13
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-2031
+---
+
+# Generic Device Plugin for MicroShift
+
+## Summary
+
+[Generic Device Plugin](https://github.com/squat/generic-device-plugin) is a
+solution for attaching devices not needing special initialization
+(e.g. serial, USB, certain video devices) to containers. The goal of this enhancement is to
+outline its adoption for MicroShift.
+
+## Motivation
+
+MicroShift is built for edge devices, which commonly use peripherals
+like sensors and adapters.
+Currently, MicroShift does not provide any solution to pass these peripherals
+through to Pods without requiring elevated privileges.
+We want to provide a solution for attaching simple devices (not requiring any special
+initialization) to Pods without potentially compromising on security.
+
+### User Stories
+
+* As a MicroShift device administrator, I want to attach simple devices to Pods
+  without elevating privileges (e.g. mounting `/dev`, adding extraneous capabilities).
+
+### Goals
+
+* Include functionality of
+  [Generic Device Plugin](https://github.com/squat/generic-device-plugin)
+  in MicroShift
+
+### Non-Goals
+
+* Adapt Generic Device Plugin (e.g. create Operator) for OpenShift or other Kubernetes distribution
+
+## Proposal
+
+* Import Generic Device Plugin code into MicroShift binary
```

Comment:
How do we account for CVEs and compliance issues if we add that code?

Reply:
Work with upstream on fixing those. I guess fixing CVEs is beneficial for everyone.